### ISSUE TITLE
Text colors are wrong

### ISSUE THREAD

**Issue by Marie-Joseph** [Dgame/Dgame#78]
When setting foreground or background colors for Text objects, the colors are rendered as though they were read as the BGR values rather than the RGB values. For example, a Text object with the foreground color Color4b.Yellow (0xFFFF00) will actually render as cyan (0x00FFFF), and Color4b.Cyan will render as yellow.

From what I can tell, this happens no matter how colors are provided, whether using the Color4b default values, hex values, or sets of traditional 0-255 RGB values. Through experimentation, I've confirmed it's an RGB to BGR switch.

This does not happen elsewhere.

Software versions:
dmd 2.092.0
Dgame 0.6.7
derelict-util 2.1.0
derelict-gl3 1.0.24
derelict-sdl2 1.9.5
Derelict loads GL 45 (I think that means 4.5)
SDL2 2.0.12

Someone else has already confirmed this on a completely different machine building from the same dub repository with the same compiler. That said, here's some other potentially relevant system information:
OS: Fedora 32
Driver: Mesa 20.1.4
APU: Ryzen 7 3700u w/ Vega 8 Raven graphics

**Comment by Dgame** [comment]
Have you tried another mode, like `Solid`, `Shaded` and `Blended`? It can be different depending on the mode. Since you say that you have never experienced this behaviour anywhere else, the `_transfer` function, which translates between `Color4b` and `SDL_Color`, is probably fine. This would be my first guess otherwise. Could you provide me with a code example and the resulting screenshot?

**Comment by Marie-Joseph** [comment]
Indeed, I have tried it with all three modes to see if that was the problem. Should have mentioned that.

There's a lot of places in the actual code where I use Text objects, but the below is representative of them all:
```
titleText = new Text(titleFont, "Bladjad");
titleText.mode = Font.Mode.Shaded;
titleText.foreground = Color4b.Cyan;
titleText.background = Color4b(0x143D4C);
titleText.update();
titleText.setPosition((WndDim.width / 2) - (titleText.width / 2),
                       WndDim.height / 5);
```

The result:
![Screenshot from 2020-07-31 03-08-33](https://user-images.githubusercontent.com/47213432/89014848-bdab1c80-d2db-11ea-8d73-86ae9e440109.png)

Note that the clear color is set by `wnd.setClearColor(Color4b(0x4C3D14))`. This font doesn't seem to respect opacity so I can't make the text background transparent.

Actually, that reminds me... I also can't change Text objects after the first time they're rendered, it seems... This may not be a bug, but I thought I'd mention it in case it is. Especially in case it's related.
